wcfsetup/install/files/lib/system/search/mysql/MysqlSearchEngine.class.php

```diff
@@ -59,7 +60,11 @@ public function search(
             }
             $additionalConditionsConditionBuilder = ($additionalConditions[$objectTypeName] ?? null);
 
-            $query = $objectType->getOuterSQLQuery($q, $searchIndexCondition, $additionalConditionsConditionBuilder);
+            if ($objectType instanceof ISearchProvider) {
+                $query = $objectType->getOuterSqlQuery($additionalConditionsConditionBuilder);
+            } else {
+                $query = $objectType->getOuterSQLQuery($q, $searchIndexCondition, $additionalConditionsConditionBuilder);
+            }
```

Comment:
This is not ideal, because it breaks existing search engines (e.g. Elasticsearch). Perhaps the `getOuterSqlQuery()` method in the ISearchProvider could be renamed to something more fitting [1] and then a `->getOuterSQLQuery()` method be added to the abstract parent class that provides the backwards compatibility with old search engines?

[1] getFetchObjectsQuery() perhaps?